### PR TITLE
[RFR] Sanitize AutocompleteInput

### PR DESCRIPTION
**Pull request by tkvw**
I tried to set the `fullWidth` attribute on `AutoCompleteInput` and expected it to work like all other Inputs does, but it didn't work. So I sanitized the component and passed through the props to bring it inline with the other input components.

### CODE REVIEW

**Comment by fzaninotto** [comment]
Can you rebase?

**Comment by tkvw** [comment]
Done.

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js [hunk 1]
```diff
@@ -281,22 +299,13 @@ export class AutocompleteInput extends React.Component {
...
-                margin="normal"
-                className={classnames(classes.root, className)}
-                inputRef={ref}
                 error={!!(touched && error)}
                 helperText={(touched && error) || helperText}
+                className={classnames(classes.root, className)}
                 {...options}
-                InputProps={{
-                    classes: {
-                        input: classes.input,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.js]: You're breaking existing code by removing that

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js [hunk 2]
```diff
@@ -281,22 +299,13 @@ export class AutocompleteInput extends React.Component {
...
-                    },
-                    ...InputProps,
-                    ...other,
-                }}
+                {...input}
+                {...sanitizeRestProps(rest)}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.js]: In the previous code, the rest props are passed to the Input, now they are passed to the TextField.

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js [hunk 3]
```diff
@@ -281,22 +299,13 @@ export class AutocompleteInput extends React.Component {
...
+                className={classnames(classes.root, className)}
                 {...options}
-                InputProps={{
-                    classes: {
-                        input: classes.input,
-                    },
-                    ...InputProps,
-                    ...other,
-                }}
+                {...input}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.js]: why did you add these props?

File: packages/ra-ui-materialui/src/input/ReferenceInput.js [hunk 4]
```diff
@@ -54,6 +54,7 @@ export const ReferenceInputView = ({
     classes,
     className,
     error,
+    fullWidth,
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/input/ReferenceInput.js]: This is exactly the type of props that we want to pass with `rest` instead of identifying them individually here.

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js (hunk 3 above)
Comment (tkvw) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.js]: I brought the component inline with `TextInput`, which is better designed. Same applies for this component.

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js (hunk 2 above)
Comment (tkvw) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.js]: Correct, imo previous was wrong.

File: packages/ra-ui-materialui/src/input/AutocompleteInput.js (hunk 1 above)
Comment (tkvw) [reply packages/ra-ui-materialui/src/input/AutocompleteInput.js]: The `classes` prop is still passed through, but `classes.input` doesn't make sense (there's no `classes.input` in the styles of `AutocompleteInput`). Not sure why it was there in the first place.

File: packages/ra-ui-materialui/src/input/ReferenceInput.js (hunk 4 above)
Comment (tkvw) [reply packages/ra-ui-materialui/src/input/ReferenceInput.js]: I did this, because of the `rest` props were still being passed to `Labeled`, this is no long the case so this can be changed.

**Comment by tkvw** [comment]
So what I did was to make the `TextField` generation inline with `TextInput` and added a `propsSanitizer`.

**Comment by fzaninotto** [comment]
This PR was closed by GitHub when I renamed the repository from "admin-on-rest" to "react-admin", and the `next` branch `master`.

Would you mind reopening a new one with the same changes?
